### PR TITLE
Add mutation observer

### PR DESCRIPTION
**Pull request by peterszerzo**
- add a mutation observer instance.
 - use it to run URL change logic.
 - use it to run highlights more efficiently (remove setInterval).

### CODE REVIEW

File: packages/journey-manager/src/index.ts [hunk 1]
```diff
@@ -445,25 +472,36 @@ export const run = async (props: RunProps): Promise<RunOutput> => {
...
+   */
+
+  let previousUrl = window.location.toString();
+
+  const documentObserver = new MutationObserver(() => {
+    debouncedSetHighlights();
+    // If the document changed for any reason (click, popstate event etc.), check if the URL also changed
+    // If it did, handle page load events
+    const newUrl = window.location.toString();
+    if (newUrl !== previousUrl) {
```
[the remark this answers is not in the document]
Comment (peterszerzo) [reply packages/journey-manager/src/index.ts:494]: You're right, this is not robust at all... I think you're very close to the right logic, the way I would formulate it is that a load step should only fire if it was not applicable on `previousUrl` but is now applicable on `newUrl`. I have a feeling you're saying the same thing, maybe just confirm here so I know how to implement?

[the remark this answers is not in the document]
Comment (peterszerzo) [reply packages/journey-manager/src/index.ts:510]: @jakub-nlx I think I have a solution, thanks for the comment!

**Review by jakub-nlx — APPROVED** [review]

File: packages/journey-manager/src/index.ts [hunk 2]
```diff
@@ -317,17 +333,35 @@ export const run = async (props: RunProps): Promise<RunOutput> => {
...
-      return;
-    }
-    sendStep(stepId, once ?? false);
-  });
+  let previousUrl = window.location.toString();
+  /**
+   * Keeps track of which load steps matched the URL the last time the URL was checked.
+   * This does not necessarily mean that the steps in question have actually been triggered (page load events should not fire if subsequent pages also satisfy the URL condition for which a step was already fired).
+   */
+  let previouslyMatchedLoadSteps: string[] = [];
```
[the remark this answers is not in the document]
Comment (peterszerzo) [reply packages/journey-manager/src/index.ts:341]: Definitely, opened https://github.com/nlxai/sdk/issues/113 to track.

File: packages/journey-manager/src/index.ts [hunk 3]
```diff
@@ -317,17 +333,35 @@ export const run = async (props: RunProps): Promise<RunOutput> => {
...
+
+  // Checks load steps to be triggered at the current URL
+  // Returns the step ID's that actually trigger so it can be compared to subsequent calls
```
[the remark this answers is not in the document]
Comment (peterszerzo) [reply packages/journey-manager/src/index.ts:344]: Oh yeah, great catch, the comment got stuck in a previous iteration of the logic.

**Review by gampleman — APPROVED** [review]

**Pull request merged**